pallets/phala/src/stakepool.rs

```diff
@@ -327,6 +327,25 @@ pub mod pallet {
 		PoolWhitelistStakerRemoved { pid: u64, staker: T::AccountId },
 		/// A worker is reclaimed from the pool
 		WorkerReclaimed { pid: u64, worker: WorkerPublicKey },
+		/// Shares that a staker gained when he makes contribution
+		ContributionShare {
+			pid: u64,
+			user: T::AccountId,
+			share: BalanceOf<T>,
+		},
+		/// Shares that a staker droped when he makes withdrawal
+		WithdrawalShare {
+			pid: u64,
+			user: T::AccountId,
+			share: BalanceOf<T>,
+		},
+		/// The amount of reward that distributed to owner and stakers
+		DistributeReward {
+			pid: u64,
+			owner: T::AccountId,
+			commission: BalanceOf<T>,
+			distribution: BalanceOf<T>,
+		},
```

Comment:
```suggestion
		/// A staker has contributed to a pool and got some shares
		Contributed {
			pid: u64,
			user: T::AccountId,
			shares: BalanceOf<T>,
		},
		/// Shares that a staker droped when he makes withdrawal
		ShareWithdrawn {
			pid: u64,
			user: T::AccountId,
			shares: BalanceOf<T>,
		},
		/// Some rewards have been received by a pool and is about to distribute
		RewardReceived {
			pid: u64,
			to_owner: BalanceOf<T>,
			to_stakers: BalanceOf<T>,
		},
```

Btw, I think `RewardReceived::owner` is not required, since you always know the owner of a pool